### PR TITLE
Docs: Plugin-E2E: Docs improvements

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

@mckn have read through the `Get started` docs, and he gave me the following feedback: 
* We should have `You will learn` summary at the top of the page (similar to the [Playwright docs](https://playwright.dev/docs/intro#introduction))
* Move part about `.gitignore` from Installing Playwright section to Configure Playwright
* Move part about removing files from Installing Playwright section to Writing tests

All these bullet point should be addressed in this PR.

Furthermore, he suggested to drop the idea of having examples based on a scaffolded backend data source, and instead simply test that auth (the config) works. So instead of having plugin type specific tests in the Get started guide, we provide links to the following guides: 
* How to test a panel plugin
* How to test a data source plugin
* How to test an app plugin

I like this idea. However, since we don't have docs for any of these plugin types yet, I think we should keep the data source specific example for now as it, in only a few lines of code, showcases what plugin-e2e is capable of. Once we have published docs for how to test the different plugin types, we can update the Get started guide again any make it plugin type agnostic. Thoughts @mckn? 

Aside from that, this PR includes the following changes:
* Adds the `introduction` page to the sidebar menu. This content on this page is information, and I'm concerned people would miss out on it if it's not present in the sidebar menu.
* Align content in the `authentication.md` article with the latest changes in the get-started guide. 
* Remove the code formatting from all headers as it looks weird
 
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by josmperez** [comment]
> * Remove the code formatting from all headers as it looks weird

By point of style, Grafana follows Google's style recommendation which is to permit code formatting in headers so long as it is surrounded by descriptive text. So, `MethodName` is not permitted, but 'The `MethodName` method" is permitted. 

I can see making an exception for this in the case of the name of a tool which is also a CLI command.  For example, we could say either "Use the create-plugin tool" but "Use the `create-plugin` command on the terminal".

File: docusaurus/docs/e2e-test-a-plugin/authentication.md [hunk 1]
```diff
@@ -19,54 +19,52 @@ To be able to interact with the Grafana UI, you need to be logged in to Grafana.
 
 ### Plugins that don't use RBAC
 
-If your plugin doesn't use RBAC, you can use the default server administrator credentials to login. The following snippet is a [setup project](https://playwright.dev/docs/test-global-setup-teardown#setup-example) that invokes a function in the `@grafana/plugin-e2e` package that will login to Grafana using `admin:admin`. The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`.
+If your plugin doesn't use RBAC, you can use the default server administrator credentials to login. In the example below, there's a [setup project](https://playwright.dev/docs/test-global-setup-teardown#setup-example) called `auth`. This project invokes a function in the `@grafana/plugin-e2e` package that will login to Grafana using `admin:admin`. The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/authentication.md:22]:
```suggestion
If your plugin doesn't use RBAC, you can use the default server administrator credentials to login. In the following example, there's a [setup project](https://playwright.dev/docs/test-global-setup-teardown#setup-example) called `auth`. This project invokes a function in the `@grafana/plugin-e2e` package that will login to Grafana using `admin:admin`. The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`.
```

Comment (josmperez) [reply docusaurus/docs/e2e-test-a-plugin/authentication.md:22]: This is a nit, but style guidelines generally prefer "following example" to "example below" because it does not depend on a specific spacial relationship. Theoretically different devices could display content "below" in different ways, not all of which are spacially below.

File: docusaurus/docs/e2e-test-a-plugin/authentication.md [hunk 2]
```diff
@@ -19,54 +19,52 @@ To be able to interact with the Grafana UI, you need to be logged in to Grafana.
...
-// playwright.config.js|ts
-projects: [
-  {
-    name: 'auth',
-    testDir: 'node_modules/@grafana/plugin-e2e/dist/auth',
-    testMatch: [/.*\.js/],
-  },
-];
-```
+The second project, `run-tests`, runs all tests in the `./tests` directory. This project reuses the authentication state from the `auth` project. This means login will only happen once, and all tests in the `run-tests` project will start already authenticated.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/authentication.md:24]:
```suggestion
The second project, `run-tests`, runs all tests in the `./tests` directory. This project reuses the authentication state from the `auth` project. This means login only happens once, and all tests in the `run-tests` project will start already authenticated.
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 3]
```diff
@@ -29,9 +29,14 @@ import ScaffoldPluginE2EDSWorkflowNPM from '@snippets/plugin-e2e-ds-workflow.npm
 import ScaffoldPluginE2EDSWorkflowYarn from '@snippets/plugin-e2e-ds-workflow.yarn.md';
 import ScaffoldPluginE2EDSWorkflowPNPM from '@snippets/plugin-e2e-ds-workflow.pnpm.md';
 
-# Get started
+This guide will walk through how to get started with end-to-end testing in your plugin.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:32]:
```suggestion
This guide walks you through how to get started with end-to-end testing in your plugin.
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 4]
```diff
@@ -29,9 +29,14 @@ import ScaffoldPluginE2EDSWorkflowNPM from '@snippets/plugin-e2e-ds-workflow.npm
...
 
-This article will guide through how to install and configure `@grafana/plugin-e2e`, write tests and setup a basic Github workflow that can run your end-to-end tests targeting multiple versions of Grafana.
+**You will learn:**
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:34]:
```suggestion
**Topics include:**
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 5]
```diff
@@ -29,9 +29,14 @@ import ScaffoldPluginE2EDSWorkflowNPM from '@snippets/plugin-e2e-ds-workflow.npm
...
+
+- how to install `@grafana/plugin-e2e`
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:36]:
```suggestion
- How to install `@grafana/plugin-e2e`
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 6]
```diff
@@ -29,9 +29,14 @@ import ScaffoldPluginE2EDSWorkflowNPM from '@snippets/plugin-e2e-ds-workflow.npm
...
+- how to configure authentication
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:37]:
```suggestion
- How to configure authentication
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 7]
```diff
@@ -29,9 +29,14 @@ import ScaffoldPluginE2EDSWorkflowNPM from '@snippets/plugin-e2e-ds-workflow.npm
...
+- how to write tests
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:38]:
```suggestion
- How to write tests
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 8]
```diff
@@ -29,9 +29,14 @@ import ScaffoldPluginE2EDSWorkflowNPM from '@snippets/plugin-e2e-ds-workflow.npm
...
+- how to setup a basic Github workflow that runs tests against multiple versions of Grafana
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:39]:
```suggestion
- How to setup a basic Github workflow that runs tests against multiple versions of Grafana
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 9]
```diff
@@ -45,15 +50,9 @@ This article will guide through how to install and configure `@grafana/plugin-e2
...
 ### Installing Playwright
 
-Please refer to the [Playwright documentation](https://playwright.dev/docs/intro#installing-playwright) for instruction on how to install. `@grafana/plugin-e2e` extends Playwright APIs, so you need to have `Playwright/test` with a minimum version of 1.41.2 installed as a dev dependency in the package.json file of your plugin.
-
-You can remove any test files (`[filename].spec.js|ts`) that was generated when you installed Playwright. Let's also add the following line to your `.gitignore` file.
-
-```shell title=".gitignore"
-/playwright/.auth/
-```
+`@grafana/plugin-e2e` extends Playwright APIs, so you need to have `@playwright/test` with a minimum version of 1.41.2 installed as a dev dependency in the package.json file of your plugin. Please refer to the [Playwright documentation](https://playwright.dev/docs/intro#installing-playwright) for instruction on how to install. Make sure you can run the example tests that were generated when you installed Playwright before you proceed to the next step in this guide.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:53]:
```suggestion
The `@grafana/plugin-e2e` tool extends Playwright APIs, so you need to have `@playwright/test` with a minimum version of 1.41.2 installed as a dev dependency in the `package.json` file of your plugin. Refer to the [Playwright documentation](https://playwright.dev/docs/intro#installing-playwright) for instruction on how to install Playwright. Make sure you can run the example tests that were generated when you installed Playwright before you proceed to the next step in this guide.
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 10]
```diff
@@ -45,15 +50,9 @@ This article will guide through how to install and configure `@grafana/plugin-e2
...
 
-## Set up `plugin-e2e`
+## Set up @grafana/plugin-e2e
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:55]:
```suggestion
## Set up `@grafana/plugin-e2e`
```

Comment (josmperez) [reply docusaurus/docs/e2e-test-a-plugin/get-started.md:55]: See comment in GitHub conversations regarding this. Grafana style, following Google, prefers `code formatting` in headers so long as a code term is not used alone.

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 11]
```diff
@@ -124,6 +123,12 @@ export default defineConfig({
 });
 ```
 
+The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`. To prevent these files from being version controlled, you can add the following line to your `.gitignore` file.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:126]:
```suggestion
The authenticated state is stored on disk and the file name pattern is as follows: `<plugin-root>/playwright/.auth/<username>.json`. To prevent these files from being version controlled, you can add the following line to your `.gitignore` file:
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 12]
```diff
@@ -176,9 +183,9 @@ queryString="current-package-manager"
 
 ### Step 6: Run tests in CI
 
-We recommend using a CI workflow to run end-to-end tests to continuously check for breakages. The following Github workflow will run end-to-end tests against a range of Grafana versions for every PR in your Github repository. 
+We recommend using a CI workflow to run end-to-end tests to continuously check for breakages. The following Github workflow will run end-to-end tests against a range of Grafana versions for every PR in your Github repository.
 
-:::note 
+:::note
 This is a generic example based on a backend plugin. You may need to alter or remove some of the steps in the `playwright-tests` job before using it in your plugin.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:189]:
```suggestion

This is a generic example based on a backend plugin. You may need to alter or remove some of the steps in the `playwright-tests` job before using it in your plugin.

```

File: docusaurus/docs/e2e-test-a-plugin/setup-resources.md [hunk 13]
```diff
@@ -65,7 +65,7 @@ If running end-to-end tests in CI requires provisioning, you may need to remove
 
 `@grafana/plugin-e2e` provides fixtures that enables you to read files that you have placed in the `provisioning` folder.
 
-#### `readProvisionedDataSource` fixture
+#### readProvisionedDataSource fixture
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/setup-resources.md:68]:
```suggestion
#### The `readProvisionedDataSource` fixture
```

File: docusaurus/docs/e2e-test-a-plugin/setup-resources.md [hunk 14]
```diff
@@ -79,7 +79,7 @@ const datasource = readProvisionedDataSource({ fileName: 'datasources.yml' });
 await panelEditPage.datasource.set(datasource.name);
 ```
 
-#### `readProvisionedDashboard` fixture
+#### readProvisionedDashboard fixture
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/setup-resources.md:82]:
```suggestion
#### The `readProvisionedDashboard` fixture
```

**Review by josmperez — APPROVED** [review]
Some changes recommended, but no blocking issues. LGTM.

**Comment by sunker** [comment]
Thanks for feedback @josmperez!

Regarding code formatting in headers - nice did not know this was permitted. The thing is that with current docusaurus styling, it's looks pretty weird having two fonts in headers. See this discussion [here](https://github.com/grafana/plugin-tools/pull/748/files/27166884a838011feb79447d831d59e3f6684144#r1491374421). I don't have strong opinions here, so I'm happy to change it back to use code formatting if you want. But in that case, I should probably do it in a separate PR that changes all headers for all plugin-e2e docs, as all other articles don't use code formatting in headers right now.

**Comment by mckn** [comment]
> I like this idea. However, since we don't have docs for any of these plugin types yet, I think we should keep the data source specific example for now as it, in only a few lines of code, showcases what plugin-e2e is capable of. Once we have published docs for how to test the different plugin types, we can update the Get started guide again any make it plugin type agnostic. Thoughts @mckn?

Sounds good! Lets push that change forward!

**Review by mckn — APPROVED** [review]
LGTM!

I would love a section at the end of the getting started called "where to go next?" where we add links to next guides/pages of documentation to read when you are done with the getting started.

**Comment by sunker** [comment]
> I would love a section at the end of the getting started called "where to go next?" where we add links to next guides/pages of documentation to read when you are done with the getting started.

Yeah me too! Let's add that when we have published guides on how to test app plugins, ds plugin and panel plugins.
